includes/course-builder/class-sensei-course-outline-block.php

```diff
@@ -0,0 +1,64 @@
+<?php
+/**
+ * File containing the class Sensei_Course_Outline_Block.
+ *
+ * @package sensei
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Class Sensei_Course_Outline_Block
+ */
+class Sensei_Course_Outline_Block {
+	/**
+	 * Register course outline block.
+	 */
+	public function register_block_type() {
+		register_block_type(
+			'sensei-lms/course-outline',
+			[
+				'render_callback' => [ $this, 'render' ],
+				'editor_script'   => 'sensei-course-builder',
+				'attributes'      => [
+					'course_id' => [
+						'type' => 'int',
+					],
+				],
+				'supports'        => [],
+			]
+		);
+	}
+
+	/**
+	 * Render course outline block.
+	 *
+	 * @param array  $attributes
+	 * @param string $content
+	 *
+	 * @return string
+	 */
+	public function render( $attributes, $content ) {
+
+		global $post;
+
+		$templates = explode( '## template ##', $content );
+
+		$loader = new \Twig\Loader\ArrayLoader( [] );
+		$twig = new \Twig\Environment( $loader );
```

Comment:
Probably it's not the best way to create this instance because we don't need a loader. But I didn't dig into this at this moment to figure out the best approach.